### ISSUE TITLE
bbmri link broken

### ISSUE THREAD

**Issue by sergiocontrino** [ecrin-github/MDR_FuiPortal#55]
there is a change of format for the bbmri link

from vittorio 

> here is an example of an old collection URL and the corresponding new one:
old: [https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:NL_AMCBB:collection:AB21-022](https://fra01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdirectory.bbmri-eric.eu%2FERIC%2Fdirectory%2F%23%2Fcollection%2Fbbmri-eric%3AID%3ANL_AMCBB%3Acollection%3AAB21-022&data=05%7C02%7CSergio.CONTRINO%40Ecrin.org%7Cb917877e48bc4893f81d08dce450d98f%7C7d5912758f7c4476bb141f96f2587607%7C0%7C0%7C638636283154619402%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=pom4lfj0kL0gezd7jFaryBE1%2B9GPRbBOZEXKSgIgcOs%3D&reserved=0)
new: [https://directory.bbmri-eric.eu/ERIC/directory/#/collection/bbmri-eric:ID:NL_AMCBB:collection:AB21-022](https://fra01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdirectory.bbmri-eric.eu%2FERIC%2Fdirectory%2F%23%2Fcollection%2Fbbmri-eric%3AID%3ANL_AMCBB%3Acollection%3AAB21-022&data=05%7C02%7CSergio.CONTRINO%40Ecrin.org%7Cb917877e48bc4893f81d08dce450d98f%7C7d5912758f7c4476bb141f96f2587607%7C0%7C0%7C638636283154641827%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=feI59e%2FqsSL%2Byy5i28Gd%2B38ChW9Wk7NPbZh4jHKauRk%3D&reserved=0)
Basically, the difference is the addition of "/ERIC/directory/".

**Comment by sergiocontrino** [comment]
the links are stored in
search.new_objects
and 
search.objects

e.g.
`select url from search.new_objects where url like '%bbmri%';

https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:UK_GBR-1-123:collection:1
https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:AT_MUG:collection:COVAC-DMStudy:diabetesmellitustype1diabetesmellitustype2COVID-19COVID-19vaccine
https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:AT_MUG:collection:EmmyStudy
https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:AT_MUG:collection:EPATHstudy
`

TODO: script to update (by replacing part of the string...)

**Comment by sergiocontrino** [comment]
https://directory.bbmri-eric.eu/#/collection/bbmri-eric:ID:UK_GBR-1-123:collection:1
->
https://directory.bbmri-eric.eu/ERIC/directory/#/collection/bbmri-eric:ID:UK_GBR-1-123:collection:1

**Comment by sergiocontrino** [comment]
tables to update:

- search.new_objects
- search.objects
- core.new_search_objects

and the corresponding json ones:

- search.new_objects_json
- search.objects_json
- core.new_search_objects_json

**Comment by sergiocontrino** [comment]
sql to be used (ex):
`update search.new_objects 
set url = replace(url, 'bbmri-eric.eu/', 'bbmri-eric.eu/ERIC/directory/')
where url like '%https://directory.bbmri-eric.eu/#/collection%';
`

and (tbc)
`update search.new_objects_json 
set full_object -> 'access_details' ->> 'url' as url replacereplace(url, 'bbmri-eric.eu/', 'bbmri-eric.eu/ERIC/directory/') where full_object -> 'access_details' ->> 'url' like '%bbmri%';
`

**Comment by sergiocontrino** [comment]
for a json column, the update stm is actually:
```
UPDATE search.new_objects_json
SET full_object = (full_object::jsonb || jsonb_build_object('access_details', 
    (full_object::jsonb -> 'access_details') || jsonb_build_object('url', replace((full_object::jsonb -> 'access_details' ->> 'url')::text, 'bbmri-eric.eu/', 'bbmri-eric.eu/ERIC/directory/'))
))::json
WHERE full_object -> 'access_details' ->> 'url' like '%bbmri%';
```

note: probably jsonb should have been used (rather than json)

**Comment by sergiocontrino** [comment]
all the stms used for the update, 284 rows in all tables

### search.new_object_json 

```
UPDATE search.new_objects_json
SET full_object = (full_object::jsonb || jsonb_build_object('access_details', 
    (full_object::jsonb -> 'access_details') || jsonb_build_object('url', replace((full_object::jsonb -> 'access_details' ->> 'url')::text, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#'))
))::json
WHERE full_object -> 'access_details' ->> 'url' like '%bbmri%';
```
### search.objects_json 
```
UPDATE search.objects_json
SET full_object = (full_object::jsonb || jsonb_build_object('access_details', 
    (full_object::jsonb -> 'access_details') || jsonb_build_object('url', replace((full_object::jsonb -> 'access_details' ->> 'url')::text, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#'))
))::json
WHERE full_object -> 'access_details' ->> 'url' like '%bbmri%';
```

### core.new_search_objects_json

```
UPDATE core.new_search_objects_json
SET full_object = (full_object::jsonb || jsonb_build_object('access_details', 
    (full_object::jsonb -> 'access_details') || jsonb_build_object('url', replace((full_object::jsonb -> 'access_details' ->> 'url')::text, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#'))
))::json
WHERE full_object -> 'access_details' ->> 'url' like '%bbmri%';
```
 
-----------

### search.new_objects
```
update search.new_objects 
set url = replace(url, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#') 
where url like '%bbmri-eric.eu/#%';
```

### search.objects
```
update search.objects 
set url = replace(url, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#') 
where url like '%bbmri-eric.eu/#%';
```

### core.new_search_objects

```
update core.new_search_objects 
set url = replace(url, 'bbmri-eric.eu/#', 'bbmri-eric.eu/ERIC/directory/#') 
where url like '%bbmri-eric.eu/#%';
```

**Comment by sergiocontrino** [comment]
this seems solved for now. there is a branch to intercept and correct the obsolete url on the UI in case necessary.
closing for now, will check with vittorio.
